rows.go

```diff
@@ -65,11 +66,19 @@ func (rows *mysqlRows) Columns() []string {
 }
 
 func (rows *mysqlRows) Close() (err error) {
+	if f := rows.finish; f != nil {
+		f()
+		rows.finish = nil
+	}
+
 	mc := rows.mc
 	if mc == nil {
 		return nil
 	}
-	if mc.netConn == nil {
+	if mc.isBroken() {
```

Comment:
I think its fine as it it. If `mc.error` is used, then we also have to check if the return value was `!= nil`. It doesn't really make the code cleaner IMO